openverse_catalog/dags/providers/provider_api_scripts/europeana.py

```diff
@@ -8,266 +8,172 @@
+            "rows": str(self.batch_limit),
+            "media": "true",
+            "start": 1,
+            "qf": ["TYPE:IMAGE", "provider_aggregation_edm_isShownBy:*"],
+            # As a dated DAG, Europeana accepts a ``query`` prop in the
+            # request params that delineates the timestamps between which
+            # records will have been added. The base class sets up the
+            # ``self.date`` attribute for us, so we can construct that
+            # ``query`` prop for the request params ahead of time.
+            "query": self._get_timestamp_query_param(self.date),
+        }
 
+        self.record_builder = EuropeanaRecordBuilder()
 
-def _extract_image_list_from_json(response_json):
-    if response_json is None or str(response_json.get("success")) != "True":
-        image_list, next_cursor, total_number_of_images = None, None, None
-    else:
-        image_list = response_json.get("items")
-        next_cursor = response_json.get("nextCursor")
-        total_number_of_images = response_json.get("totalResults")
+    def _get_timestamp_query_param(self, date):
+        date_obj = datetime.strptime(date, "%Y-%m-%d")
+        utc_date = date_obj.replace(tzinfo=timezone.utc)
+        start_timestamp = utc_date.isoformat()
+        end_timestamp = (utc_date + timedelta(days=1)).isoformat()
 
-    return image_list, next_cursor, total_number_of_images
+        start_timestamp = start_timestamp.replace("+00:00", "Z")
+        end_timestamp = end_timestamp.replace("+00:00", "Z")
 
+        return f"timestamp_created:[{start_timestamp} TO {end_timestamp}]"
 
-def _process_image_list(image_list):
-    prev_total = 0
-    total_images = 0
-    for image_data in image_list:
-        total_images = _process_image_data(image_data)
-        if total_images is None:
-            total_images = prev_total
-        else:
-            prev_total = total_images
-
-    return total_images
-
-
-def _process_image_data(image_data, sub_providers=SUB_PROVIDERS, provider=PROVIDER):
-    logger.debug(f"Processing image data: {image_data}")
-    license_url = _get_license_url(image_data.get("rights"))
-    image_url = image_data.get("edmIsShownBy")[0]
-    foreign_landing_url = _get_foreign_landing_url(image_data)
-    foreign_id = image_data.get("id")
-    title = image_data.get("title")[0]
-    meta_data = _create_meta_data_dict(image_data)
-
-    data_providers = set(meta_data["dataProvider"])
-    eligible_sub_providers = {
-        s for s in sub_providers if sub_providers[s] in data_providers
-    }
-    if len(eligible_sub_providers) > 1:
-        raise Exception(
-            f"More than one sub-provider identified for the "
-            f"image with foreign ID {foreign_id}"
-        )
-    source = (
-        eligible_sub_providers.pop() if len(eligible_sub_providers) == 1 else provider
-    )
+    def get_next_query_params(self, prev_query_params) -> dict:
+        if not prev_query_params:
+            return self.base_request_body
```

Comment:
From looking at the original code, do you not need `"cursor": "*"` in the initial request? It seems like it's working fine as is, though 🤔

Reply:
Added back in 1ea1a507